src/main/java/com/pusher/rest/data/TriggerResult.java

```diff
@@ -0,0 +1,58 @@
+package com.pusher.rest.data;
+
+import java.util.Map;
+
+import com.google.gson.Gson;
+
+public class TriggerResult {
+    static class Data {
+        public final Map<String, String> event_ids;
+
+        public Data(final Map<String, String> event_ids) {
+            this.event_ids = event_ids;
+        }
+    }
+
+    private static final Gson GSON = new Gson();
+
+    private final Result.Status status;
+    private final Integer httpStatus;
+    private final String message;
+
+    public TriggerResult(final Result result) {
+        this.status = result.getStatus();
+        this.httpStatus = result.getHttpStatus();
+        this.message = result.getMessage();
+    }
+
+    /**
+     * Get the enum classifying the result of the call
+     */
+    public Result.Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Get the HTTP status code of the call, useful for debugging instances of UNKNOWN_ERROR
+     */
+    public Integer getHttpStatus() {
+        return httpStatus;
+    }
+
+    /**
+     * Get the data response (success) or descriptive message (error) returned from the call
+     */
+    public String getMessage() {
+        return message;
+    }
+
+    /**
+     * Get the map from channel name to event ID.
+     *
+     * Response might be null.
+     */
+    public Map<String, String> getEventIDs() {
+        final Data obj = GSON.fromJson(this.message, Data.class);
```

Comment:
Use Response.getStatus (I think - the one that returns an enum, one member
of which is SUCCESS)
On 27 Mar 2015 17:28, "zimbatm" <notifications@github.com> wrote:

> In src/main/java/com/pusher/rest/data/TriggerResult.java
> <https://github.com/pusher/pusher-http-java/pull/4#discussion_r27313882>:
>
> > +    }
> > +
> > +    /**
> > +     * Get the data response (success) or descriptive message (error) returned from the call
> > +     */
> > +    public String getMessage() {
> > +        return message;
> > +    }
> > +
> > +    /**
> > +     * Get the map from channel name to event ID.
> > +     *
> > +     * Response might be null.
> > +     */
> > +    public Map<String, String> getEventIDs() {
> > +        final Data obj = GSON.fromJson(this.message, Data.class);
>
> How do we know it's a success ? We don't know if the body will contain the
> "event_ids" key until it's unmarshalled
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pusher/pusher-http-java/pull/4/files#r27313882>.
>